tests/phpunit/tests/term/query.php

```diff
@@ -451,6 +451,48 @@ public function test_object_ids_cache_should_be_invalidated_by_term_relationship
 		$this->assertSameSets( array( $terms[1] ), $found );
 	}
 
+	/**
+	 *
+	 * @ticket 63256
+	 */
+	public function test_object_ids_null_should_return_all_terms() {
+		register_taxonomy( 'wptests_tax_1', 'post' );
+
+		$terms = self::factory()->term->create_many( 3, array( 'taxonomy' => 'wptests_tax_1' ) );
+
+		$query = new WP_Term_Query(
+			array(
+				'taxonomy'   => 'wptests_tax_1',
+				'object_ids' => null,
+				'hide_empty' => false,
+				'fields'     => 'ids',
+			)
+		);
+
+		$this->assertSameSets( $terms, $query->terms );
+	}
+
+	/**
+	 *
+	 * @ticket 63256
+	 */
+	public function test_object_ids_zero_should_be_treated_as_numeric() {
+		register_taxonomy( 'wptests_tax_1', 'post' );
+
+		$term = self::factory()->term->create( array( 'taxonomy' => 'wptests_tax_1' ) );
+
+		$query = new WP_Term_Query(
+			array(
+				'taxonomy'   => 'wptests_tax_1',
+				'object_ids' => 0,
+				'hide_empty' => false,
+				'fields'     => 'ids',
+			)
+		);
+
+		$this->assertIsArray( $query->terms );
```

Comment:
The phpdoc type for `WP_Term_Query::$terms` should be updated from `array` to `array|null`. Otherwise, this assertion would not seem to make sense.

This should also have a message added to the assertion.